JamfPro/JamfPro/JamfPro.pq

```diff
@@ -271,751 +121,596 @@ UAPIResource = (baseurl as text, relativepath as text, optional query as record,
-        computer_table = Table.Combine(temp_table[response]),
-        selected_table = Table.SelectColumns(computer_table, {"id", "fonts"})
-    in
-        selected_table;
-
-computeriBeaconsImpl = (baseUrl as text, pages_table as table) as table =>
-    let
-        sections = List.Transform({
-            "iBeacons"
-        }, Text.Upper),
-        queryString = Text.Combine(List.Transform(sections, Text.Upper), "&section="),
         pageSize = "100",
 
-        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "uapi/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(_[Column1]), section = sections])[results])),
+        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "api/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(Number.From(_[Column1]) - 1), section = sections])[results])),
         computer_table = Table.Combine(temp_table[response]),
-        selected_table = Table.SelectColumns(computer_table, {"id", "ibeacons"})
+        selected_table = 
+            try Table.SelectColumns(computer_table, {"id", "applications"})
+            otherwise computer_table,
+        expand_applications = 
+            try Table.ExpandListColumn(selected_table, "applications")
+            otherwise selected_table,
+        filtered_table = 
+            try Table.SelectRows(expand_applications, each [applications] <> null)
+            otherwise expand_applications,
+        expand_applications2 = 
+            try Table.ExpandRecordColumn(filtered_table, "applications", {"name", "path", "version", "macAppStore", "sizeMegabytes", "bundleId", "updateAvailable", "externalVersionId"})
+            otherwise filtered_table
     in
-        selected_table;
-
-computerLicensedSoftwareImpl = (baseUrl as text, pages_table as table) as table =>
-    let
-        sections = List.Transform({
-            "Licensed_Software"
-        }, Text.Upper),
-        queryString = Text.Combine(List.Transform(sections, Text.Upper), "&section="),
-        pageSize = "100",
+        expand_applications2;
 
-        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "uapi/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(_[Column1]), section = sections])[results])),
-        computer_table = Table.Combine(temp_table[response]),
-        selected_table = Table.SelectColumns(computer_table, {"id", "licensedSoftware"})
-    in
-        selected_table;
 
-computerStorageImpl = (baseUrl as text, pages_table as table) as table =>
+// start of mobile device queries
+mobileDevicesImplementation = (baseUrl as text, pages_table as table) as table =>
     let
         sections = List.Transform({
-            "Storage"
+            "General",
+            "Hardware",
+            "Security"

```

Comment:
I believe the idea here that if this model works well with the rest of our changes, we can either...
1. add additional sections here
or...
2. add additional separate break-out tables like we've done for Applications below

Does that sound right?

Reply:
Yep!  My idea is one or the other based on the feedback from SAP!